### PR TITLE
Working on README and examples

### PR DESCRIPTION
**Pull request by jonathanpallant**
Trying to clarify what is HAL specific and what is RP specific.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,26 +8,26 @@
    <h3 align="center">rp-hal</h3>
 
   <p align="center">
-    A Rust HAL and board support packages for the RP family of microcontrollers from the Raspberry Pi Foundation
+    Rust support for Raspberry Pi Silicon
```
Comment (jannic) [inline README.md:11]: Isn't it better to mention that this is about microcontrollers, to avoid confusion with the traditional Raspberry Pi SBCs?

Comment (jonathanpallant) [reply README.md:11]: Fair point. I'll expand this a little bit.

**Comment by jonathanpallant** [comment]
Suggest we squash-merge this monster!

**Comment by jonathanpallant** [comment]
I'm going now. No more typing...

File: README.md [hunk 2]
```diff
@@ -37,6 +37,42 @@
 </details>
 
 <!-- GETTING STARTED -->
+
+## Getting Started
+
+So, you want to program your Raspberry Pi Silicon, using the Rust programming
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply README.md:43]: I thought it's what Eben called it, but given the PR all says "Raspberry Silicon" I probably remembered wrong. Will fix.

File: README.md [hunk 3]
```diff
@@ -147,12 +183,130 @@ RP2040 chip according to how it is connected up on the Pro Micro RP2040.
...
+Silicon device with your favourite Arm flashing/debugging tool. In addition, the
+RP2040 contains a ROM bootloader which appears as a Mass Storage Device over USB
+that accepts UF2 format images. You can use the `elf2uf2-rs` package to convert
+the Arm ELF file to a UF2 format image.
+
+For boards with USB Device support like the Raspberry Pi Pico, we recommend you
+use the UF2 process.
+
+The RP2040 contains a Cortex-M0+ processor, which implements the Thumb-2 format
+of the ARMv6-M instruction set. There are no operating-specific features in the
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply README.md:199]: I never liked the `thumbv6m` moniker - as you say, there's only one encoding so it's redundant. Will try to re-word this.

File: README.md [hunk 4]
```diff
@@ -147,12 +183,130 @@ RP2040 chip according to how it is connected up on the Pro Micro RP2040.
...
+Rust generates standard Arm ELF files, which you can load onto your Raspberry Pi
+Silicon device with your favourite Arm flashing/debugging tool. In addition, the
+RP2040 contains a ROM bootloader which appears as a Mass Storage Device over USB
+that accepts UF2 format images. You can use the `elf2uf2-rs` package to convert
+the Arm ELF file to a UF2 format image.
+
+For boards with USB Device support like the Raspberry Pi Pico, we recommend you
+use the UF2 process.
+
+The RP2040 contains a Cortex-M0+ processor, which implements the Thumb-2 format
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply README.md:198]: Fair

File: README.md [hunk 5]
```diff
@@ -147,12 +183,130 @@ RP2040 chip according to how it is connected up on the Pro Micro RP2040.
...
+[probe-run](https://github.com/knurling-rs/probe-run). This is a command-line
+tool which can flash a wide variety of microcontrollers using a wide variety of
+debug/JTAG probes. It is based on a library called
+[probe-rs](https://github.com/probe-rs/probe-rs). Unlike using, say, OpenOCD,
+probe-rs can autodetect your debug probe, which can make it easier to use.
+
+Currently, probe-rs supports the slightly unusual debug hardware in the RP2040,
+but the last released probe-run tool (v0.2.6, as of September 2021), does not.
+However, there is a special version of probe-run for the RP2040 called
+[probe-run-rs].
```
[the remark this answers is not in the document]
Comment (jonathanpallant) [reply README.md:250]: Also, the link was missing.

File: README.md [hunk 6]
```diff
@@ -40,23 +40,26 @@
...
-So, you want to program your Raspberry Pi Silicon, using the Rust programming
-language. You've come to the right place!
+So, you want to program your new Raspberry Silicon microcontroller, using the
+Rust programming language. You've come to the right place!
 
-This repository is `rp-hal` - a collection of high-level drivers for the RP2040
-and various associated boards, like the Raspberry Pi Pico and the Adafruit
-Feather RP2040.
+This repository is `rp-hal` - a collection of high-level drivers for the
+Raspberry Silicon RP2040 micontroller and various associated boards, like the
```
Comment (9names) [inline README.md:47]: missing the first r in microcontroller here

**Review by 9names — APPROVED** [review]
LGTM

**Pull request merged**